### PR TITLE
Descending lists

### PR DESCRIPTION
**Pull request by cj-xu**
We define descending lists via simultaneous definitions and show that they are isomorphic to finite multisets. The conversion from finite multisets to descending lists is exactly insertion sort. We obtain the concatenation operation on descending lists and its properties by transporting those on finite multisets.

### CODE REVIEW

**Comment by mortberg** [comment]
Can you please remove the whitespace at the end of the lines? Travis said:

```
cabal exec -- fix-agda-whitespace --check

Whitespace violation detected in /home/travis/build/agda/cubical/Cubical/Data/DescendingList/DescendingList.agda.

make: *** [check-whitespace] Error 1

The command "make test" exited with 2.
```

**Comment by mortberg** [comment]
Thanks! Let's see so that Travis is happy before merging.

Should we add the examples with sets of natural numbers as well?

**Comment by cj-xu** [comment]
> Should we add the examples with sets of natural numbers as well?

I'd love to, but I don't know where to put them. Any suggestion?

File: Cubical/Data/DescendingList/DescendingList.agda [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+open import Cubical.Data.Unit
+
+open import Cubical.Relation.Nullary
+open import Cubical.Relation.Nullary.DecidableEq
+
+open import Cubical.Foundations.Everything
+
+open import Cubical.Core.Everything
+
+open import Agda.Builtin.List using (List ; [] ; _∷_)
```
Comment (mortberg) [inline Cubical/Data/DescendingList/DescendingList.agda]: You can import https://github.com/agda/cubical/blob/master/Cubical/Data/List/Base.agda instead

**Comment by mortberg** [comment]
> > Should we add the examples with sets of natural numbers as well?
> 
> I'd love to, but I don't know where to put them. Any suggestion?

How about putting it in a nameless module in the end of the file? It's mainly there as a sanity check and for documentation.

**Comment by cj-xu** [comment]
I don't know how to instantiate module parameters inside the same module for the examples. So I instead added another file consisting of some simple examples.

File: Cubical/Data/DescendingList/DescendingList.agda (hunk 1 above)
Comment (mortberg) [reply Cubical/Data/DescendingList/DescendingList.agda]: Could you do this change please?

Comment (cj-xu) [reply Cubical/Data/DescendingList/DescendingList.agda]: I did. But I just noticed that what I did is
`open import Cubical.Data.List using (List ; [] ; _∷_)`
Or shall I change it use Base instead?

Comment (mortberg) [reply Cubical/Data/DescendingList/DescendingList.agda]: Oh I see, I guess that's fine! Thanks

**Comment by mortberg** [comment]
Ignore my previous comment. Let's just wait and see that travis is happy.

**Comment by ice1000** [comment]
Travis is happy now.

**Pull request merged**